src/test/components/ProfileCallTreeView.test.js

```diff
@@ -102,6 +103,60 @@ describe('calltree/ProfileCallTreeView', function() {
 
     expect(calltree).toMatchSnapshot();
   });
+
+  it('reacts properly to up/down navigation keys', () => {
+    function setup() {
+      // This makes the bounding box large enough so that we don't trigger
+      // VirtualList's virtualization.
+      jest
+        .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+        .mockImplementation(() => getBoundingBox(1000, 2000));
+
+      // This generates a profile where function "name<i + 1>" is present
+      // <length - i> times, which means it will have a self time of <length - i>
+      // ms. This is a good way to control the order we'll get in the call tree
+      // view: function "name1" will be first, etc.
+      const profileString = Array.from({ length: 100 }).reduce(
+        (result, func, i, array) => {
+          const funcName = `name${i + 1}  `;
+          result += funcName.repeat(array.length - i);
+          return result;
+        },
+        ''
+      );
+
+      const { profile } = getProfileFromTextSamples(profileString);
+      const store = storeWithProfile(profile);
+      const calltree = mount(
+        <Provider store={store}>
+          <ProfileCallTreeView />
+        </Provider>
+      );
+
+      return {
+        simulateKey: (key: string) =>
+          calltree.find('div.treeViewBody').simulate('keydown', { key }),
+        selectedText: () =>
+          calltree.find('.treeViewRowScrolledColumns.selected').text(),
+      };
+    }
+
+    const { simulateKey, selectedText } = setup();
```

Comment:
Optional to your opinion: I find it a bit surprising that the `setup` function is in the `it` block, usually I've written them in the describe block, then run them in the `it`.

Reply:
yeah, I know this is a bit surprising. I wanted to abstract away the possible operations but this is useful to this test only (for now).

For a test that would test "left/right" arrows, we'll need a different profile. So I'll create a "describe" block for navigation keys, and generalize a bit this `setup` function to take a profile string. I think this will be better, thanks for the comment.